File: CS-Examples/03_Cells/FindCellsWithStyleName.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace FindCellsWithStyleName
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Load the document from disk
            Workbook workbook = new Workbook();
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\SampleB_2.xlsx");

            //Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            //Get the cell style name
            string styleName = sheet.Range["A1"].CellStyleName;

            CellRange ranges = sheet.AllocatedRange;
            foreach (CellRange cc in ranges)
            {
                //Find the cells which have the same style name
                if (cc.CellStyleName == styleName)
                {
                    //Set value
                    cc.Value = "Same style";
                }
            }

            // Specify the filename for the resulting Excel file
            string result = "FindCellsWithStyleName_result.xlsx";

            // Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/03_Cells/FindCellsWithStyleName.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace FindCellsWithStyleName
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Instantiate a new workbook object.
            Dim workbook As New Workbook()

            'Load an existing Excel document from the specified file path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\SampleB_2.xlsx")

            'Retrieve the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Retrieve the style name of the cell A1.
            Dim styleName As String = sheet.Range("A1").CellStyleName

            'Retrieve the range of cells that contain data in the worksheet.
            Dim ranges As CellRange = sheet.AllocatedRange
            'Iterate through each cell range in the allocated range.
            For Each cc As CellRange In ranges
                'Check if the current cell range has the same style name as the one retrieved from cell A1.
                If cc.CellStyleName = styleName Then
                    'Set the value of the current cell range to "Same style".
                    cc.Value = "Same style"
                End If
            Next cc
            'Specify the file name for the output file.
            Dim result As String = "FindCellsWithStyleName_result.xlsx"

            'Save the modified workbook to the specified output file using Excel 2010 version.
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
